src/vanilla/store.ts

```diff
@@ -601,6 +581,15 @@ export const createStore = () => {
     })
     // recompute atom state
     readAtomState(atom)
+    // mount self
+    const mounted: Mounted = {
+      t: new Set(initialDependent && [initialDependent]),
+      l: new Set(),
+    }
+    mountedMap.set(atom, mounted)
+    if (import.meta.env?.MODE !== 'production') {
+      mountedAtoms.add(atom)
+    }
```

Comment:
It feels like our `mountAtom` has room for improvement (like propagating `changed` from dependencies).
It might be related with the previous comment, but not sure.